### ISSUE TITLE
Shouldn't we handle paging?

### ISSUE THREAD

**Issue by Nikoloutsos** [nerdishbynature/octokit.swift#188]
Hello 👋 

Recently was in need to fetch some data from GitHub API and I decided to use this library instead of creating my own API calls manually.
Then I realized that there is no handling for [paging](https://docs.github.com/en/rest/using-the-rest-api/using-pagination-in-the-rest-api?apiVersion=2022-11-28) by running the following example.

```swift
let issues = try await Octokit(config).issues(owner: "freeCodeCamp", repository: "freeCodeCamp")
// returns 100 issues while it should return 188.
```

I think it would be nice enhancement for this library to give the option to the developer to choose if he wants to get all content.

What do you say?

**Comment by Nikoloutsos** [comment]
Just saw we give the ability to add `page` on query params. But I think the developer experience would be much better if we had something like:

```swift
// Do while loop until a page returns no issues.
var issues = [Issue]()
var starting_page = 1
while(true) {
issues.append(try await Octokit(config).issues(owner: "freeCodeCamp", repository: "freeCodeCamp", page: starting_page))
starting_page += 1
}
```

instead could be something more developer friendly:
```swift
issues.append(try await Octokit(config).issues(owner: "freeCodeCamp", repository: "freeCodeCamp", paging_strategy: .allContent))

// or if you want only the first page
issues.append(try await Octokit(config).issues(owner: "freeCodeCamp", repository: "freeCodeCamp", paging_strategy: .page(number: 1, max_items:100)))
```

**Comment by pietbrauer** [comment]
I agree that in your context this would make sense. In my opinion the application should handle this though instead of `Octokit.swift`.
The main reason is that in all other swift contexts (iOS, iPadOS, visionOS, macOS) you might have a table of contents and handle the paging based on the user scrolling the contents.

I am open to discussion here as this seems still possible with your proposed solution. If you are willing you could open a PR and we'll have a look. Ideally this would be handled somewhere central … .

**Comment by Nikoloutsos** [comment]
@pietbrauer I see what you mean. What do you mean by "somewhere central"?

**Comment by pietbrauer** [comment]
Central in a way that we have one method that handles the paging in OctoKit.swift so that every API Call method can use it without having to worry which paging option is used.

**Comment by pietbrauer** [comment]
Closed via #206
